### ISSUE TITLE
Cannot upload a folder to a blob container

### ISSUE THREAD

**Issue by Melebius** [microsoft/AzureStorageExplorer#4131]
**Storage Explorer Version**: 1.17.0
**Build Number**: 20201211.10
**Platform/OS**: Windows 10 Pro
**Architecture**: i86

## Bug Description ##
I cannot upload a folder from my computer to the Azure Storage blob container. The upload is shown as finished successfully (green tick).

## Steps to Reproduce ##
1. Open a blob container in the Azure Storage Explorer.
2. Drag a local folder into the Azure Storage Explorer window.

## Expected Experience ##
The folder and all the contained files should be uploaded to the Azure storage (or an error should be shown if this is not supported).

## Actual Experience ##
Nothing is uploaded. In the Activities section, the upload is marked as successful (green tick). The following message is shown:

> Transfer of 'D:\path\subfolder-to-upload\' to '$web/designated-superfolder/' complete: 0 items transferred (used SAS, discovery completed)
> Started at: 24. 2. 2021 10:41:45, Duration: 2 seconds

## Additional Context ##
I’ve tried to drag from Explorer and Total Commander – no difference observed.

**Comment by Melebius** [comment]
I’ve discovered the upload works when I use hard links (`mklink /h`) to place the designated files into the folder. This is a feasible workaround for me.

**Comment by MRayermannMSFT** [comment]
We'll need to check with the AzCopy team if this is supported. We always ask AzCopy to follow symlinks.

**Comment by MRayermannMSFT** [comment]
@Melebius can you try to explain your initial setup again? I'm confused about what is linked to what. Maybe could you paste in exact repro steps for me? Like:

1. Make folder `foo/`
2. Place some number of files in `foo/`
2. `mklink foo-link-hard foo`
3. Drag and drop `foo-link-hard` into Storage Explorer

**Comment by MRayermannMSFT** [comment]
Hmmm maybe what you are doing is this?

1. Have files in various places on machine, say `file1.txt`, `file2.txt`,`file3.txt`
1. Make folder `foo/`
2. Create a soft link in `foo/` for each of the files, for example, in `foo/`: `mklink file1.txt-link <path to file1.txt>`
4. Drag and drop `foo/` into Storage Explorer

**Comment by MRayermannMSFT** [comment]
Ok ya, I think [these](https://github.com/microsoft/AzureStorageExplorer/issues/4131#issuecomment-827819218) steps are what you are doing.

**Comment by Melebius** [comment]
> Hmmm maybe what you are doing is this?
> 
>     1. Have files in various places on machine, say `file1.txt`, `file2.txt`,`file3.txt`
> 
>     2. Make folder `foo/`
> 
>     3. Create a soft link in `foo/` for each of the files, for example, in `foo/`: `mklink file1.txt-link <path to file1.txt>`
> 
>     4. Drag and drop `foo/` into Storage Explorer

Yes, this is what I was doing. Are you able to reproduce my issue?

**Comment by MRayermannMSFT** [comment]
> Yes, this is what I was doing. Are you able to reproduce my issue?

Yes I am able to repro! I've passed along the repro steps to the AzCopy team. I'll keep you up-to-date once they get back to me. Sorry this has taken a while.

**Comment by MRayermannMSFT** [comment]
Hey @Melebius, finally had a chat about this with the AzCopy team. They're going to get an item on their backlog to work on this, though it won't be done anytime soon. Definitely not for our upcoming milestone, but within this calendar year is probably possible. So for now, I'm going to move this issue to our milestone for AzCopy issues we're waiting on them for. Thanks for the patience!

**Comment by abfleishman** [comment]
Any update on this?  I want to do something similar
